### PR TITLE
Skip syncset resource deletion when UnknownObject condition is set

### PR DESCRIPTION
**Pull request by akhil-rane**

### CODE REVIEW

File: pkg/controller/syncsetinstance/syncsetinstance_controller.go [hunk 1]
```diff
@@ -500,6 +500,11 @@ func (r *ReconcileSyncSetInstance) deleteSyncSetResources(ssi *hivev1.SyncSetIns
 		itemLog := ssiLog.WithField("resource", fmt.Sprintf("%s/%s", resourceStatus.Namespace, resourceStatus.Name)).
 			WithField("apiversion", resourceStatus.APIVersion).
 			WithField("kind", resourceStatus.Kind)
+		unknownObjectSyncCondition := controllerutils.FindSyncCondition(resourceStatus.Conditions, hivev1.UnknownObjectSyncCondition)
+		if unknownObjectSyncCondition != nil {
+			itemLog.Debug("UnknownObject condition is set, skipping deletion")
```
Comment (dgoodwin) [inline pkg/controller/syncsetinstance/syncsetinstance_controller.go:505]: Info seems appropriate here.

**Comment by dgoodwin** [comment]
LGTM otherwise.

File: pkg/controller/syncsetinstance/syncsetinstance_controller.go (hunk 1 above)
Comment (akhil-rane) [reply pkg/controller/syncsetinstance/syncsetinstance_controller.go]: done

**Comment by csrwng** [comment]
lgtm

**Comment by gregsheremeta** [comment]
two lgtm's, making it official
/lgtm

**Pull request merged**